src/Illuminate/Foundation/Auth/Access/AuthorizesRequests.php

```diff
@@ -84,9 +84,9 @@ public function authorizeResource($model, $parameter = null, array $options = []
     {
         $model = is_array($model) ? implode(',', $model) : $model;
 
-        $parameter = is_array($parameter) ? implode(',', $parameter) : $parameter;
+        $parameter = $parameter ?: (Route::getCurrentRoute()->parameterNames ?: Str::snake(class_basename($model)));
```

Comment:
Maybe extract the nested conditional to another variable to make it more readable?

Reply:
I did as you said. Please check.